workspaces/marketplace/plugins/marketplace-common/src/types.ts

```diff
@@ -96,6 +118,39 @@ export interface MarketplacePluginSpec extends JsonObject {
   };
 }
 
+
+/**
+ * @public
+ */
+export type FullTextFilter = {
+  term: string;
+  fields?: string[];
+};
+
+/**
+ * @public
+ */
+export type GetPluginsRequest = {
+  limit?: number;
+  offset?: number;
+  filter?: Record<string, string>;
```

Comment:
we should keep this aligned with catalog types
```suggestion
  filter?: EntityFilterQuery;
```